### PR TITLE
Remove hard-coded include paths from build for --with-system-hidapi

### PR DESCRIPTION
**Pull request by markubiak**
Allows unmodified build with bitbake tools, which check to ensure no host paths are used at compile time.

### CODE REVIEW

File: chid.pxd [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from libc.stddef cimport wchar_t, size_t
 
-cdef extern from "hidapi.h":
+cdef extern from "hidapi/hidapi.h":
```
Comment (Youw) [inline chid.pxd:3]: conceptually - this isn't right
one should always include `<hidapi.h>` in the source code
and at the build/configuration level a correct include path should be added using one of the package search tools

**Comment by mcuee** [comment]
Yes hidapi should be detected using pkg-config when it is available. 

Reference:
* https://github.com/trezor/cython-hidapi/issues/133#issuecomment-1542998532

**Comment by prusnak** [comment]
> Yes hidapi should be detected using pkg-config when it is available.

Would you like to create a pull request that does this? If you do, we'll obsolete this one and merge yours instead.

**Comment by mcuee** [comment]
> > Yes hidapi should be detected using pkg-config when it is available.
> 
> Would you like to create a pull request that does this? If you do, we'll obsolete this one and merge yours instead.

Sorry but I am not a good candidate to do this. I am not a programmer myself and my involvements with open source projects concentrate on testing and technical support (libusb, hidapi, libusb-win32, libusbK, pyusb, libusbdotnet, avrdude, etc).

@Youw 
Would you help here if you got the time? Thanks.

Or @jonasmalacofilho 
Maybe you can help as well if you got the time. Thanks.

**Comment by mcuee** [comment]
This may help.
https://github.com/matze/pkgconfig/

```
$ python
Python 3.10.10 (main, Feb 10 2023, 19:20:41)  [GCC 12.2.0 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkgconfig
>>> pkgconfig.cflags('hidapi')
'-IC:/msys64/mingw64/include/hidapi'
>>> pkgconfig.libs('hidapi')
'-lhidapi'
>>> pkgconfig.parse('hidapi')
defaultdict(<class 'list'>, {'include_dirs': ['C:/msys64/mingw64/include/hidapi'], 'libraries': ['hidapi']})
>>> quit()
```

**Comment by Youw** [comment]
I can, it is only question of time.
Will add it to my TODO list.

**Comment by mcuee** [comment]
@Youw 

Just wondering if you can take a look at this again when you got some time. Thanks.

I think this might help Issue #133 as well.
* #133

**Comment by Youw** [comment]
> Just wondering if you can take a look at this again when you got some time. Thanks.

Took me way more time to implement it that I was expecting, but still, #162 is ready to be reviewed/discussed.

**Comment by prusnak** [comment]
Closing this in favor of https://github.com/trezor/cython-hidapi/pull/162

**Pull request closed without merging**